src/uct/ib/mlx5/rc/rc_mlx5_iface.c

```diff
@@ -668,10 +678,11 @@ uct_rc_mlx5_iface_is_reachable_v2(const uct_iface_h tl_iface,
     /* Check hardware tag matching compatibility */
     if ((iface_addr != NULL) &&
         ((remote_type = *(uint8_t*)iface_addr) != my_type)) {
-        uct_iface_fill_info_str_buf(
-                    params, "incompatible hardware tag matching. "
-                    "%s (local) vs %s (remote)",
-                    tm_type_to_str[my_type], tm_type_to_str[remote_type]);
+        uct_iface_fill_info_str_buf(params,
+                                    "incompatible hardware tag matching. "
+                                    "%s (local) vs %s (remote)",
+                                    uct_rc_mlx5_iface_tm_type_str(my_type),
+                                    uct_rc_mlx5_iface_tm_type_str(remote_type));
```

Comment:
seems to be unrelated

Reply:
it is needed because the is_reachable test (with garbage address) causes access to invalid memory without this fix